x/irs/keeper/grpc_query_estimate_swap_in_pool.go

```diff
@@ -24,6 +24,23 @@ func (k Keeper) EstimateSwapInPool(c context.Context, req *types.QueryEstimateSw
 	if !ok {
 		return nil, types.ErrInvalidAmount
 	}
+	ptDenom := types.PtDenom(tranche)
+	if req.Denom == ptDenom {
+		if tranche.StartTime+tranche.Maturity <= uint64(ctx.BlockTime().Unix()) {
```

Comment:
What's this case? Can you add some comments on the code?

Reply:
Processing at maturity was missing.